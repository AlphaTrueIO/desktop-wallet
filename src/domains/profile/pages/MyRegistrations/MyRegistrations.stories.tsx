import { boolean, withKnobs } from "@storybook/addon-knobs";
import React from "react";

import { MyRegistrations } from "./MyRegistrations";

export default {
	title: "Profile / Pages / My Registrations",
	decorators: [withKnobs],
};

export const Default = () => {
	const isEmpty = boolean("Empty State?", false);

	const registrations = [
		{
			type: "business",
			registrations: [
				{
					agent: "OlEBank",
					businessName: "ARK Ecosystem",
					history: [],
					website: "",
					msq: true,
					repository: [],
				},
			],
		},
	];

	return <MyRegistrations registrations={isEmpty ? [] : registrations} />;
};
