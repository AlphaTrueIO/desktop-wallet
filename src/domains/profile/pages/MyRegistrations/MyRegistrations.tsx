import { SvgCollection } from "app/assets/svg";
import { Button } from "app/components/Button";
import React from "react";

import { BusinessTable } from "./components/BusinessTable";

type Props = {
	registrations?: any;
};

type RegistrationProps = {
	type: string;
	registrations: any;
};

const EmptyRegistrations = (
	<div
		data-testid="my-registrations__empty-state"
		className="flex flex-col justify-center items-center mt-4 px-10 py-20 bg-theme-background"
	>
		<SvgCollection.RegistrationsIllustration />
		<span className="mt-7 text-sm text-theme-neutral-600">
			Register Business, Bridgechain and Delegate in the most convenient way.
		</span>
	</div>
);

const renderRegistration = ({ type, registrations }: RegistrationProps) => {
	switch (type) {
		case "business":
			return <BusinessTable key={type} data={registrations} />;

		default:
			return null;
	}
};

export const MyRegistrations = ({ registrations }: Props) => {
	console.log({ registrations });
	const mountRegistrations = () =>
		registrations.map((registrationsBlock) => {
			return renderRegistration(registrationsBlock);
		});

	return (
		<section className="bg-theme-neutral-200">
			<div className="px-10 py-14 bg-theme-background">
				<div className="flex justify-between">
					<div className="flex flex-col">
						<span className="font-semibold text-3xl">My Registrations</span>
						<span className="text-theme-neutral-600">
							You can register a Delagate, Business and Bridgechain.
						</span>
					</div>
					<Button className="self-end">Register</Button>
				</div>
			</div>
			{!registrations.length ? EmptyRegistrations : mountRegistrations()}
		</section>
	);
};

MyRegistrations.defaultProps = {
	registrations: [],
};
