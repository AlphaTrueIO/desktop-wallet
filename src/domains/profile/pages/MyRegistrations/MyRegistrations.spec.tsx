import { render } from "@testing-library/react";
import React from "react";

import { MyRegistrations } from "./MyRegistrations";

describe("Welcome", () => {
	it("should render empty state", () => {
		const { getByTestId, asFragment } = render(<MyRegistrations />);

		expect(asFragment()).toMatchSnapshot();
		expect(getByTestId("my-registrations__empty-state")).toBeTruthy();
	});
});
