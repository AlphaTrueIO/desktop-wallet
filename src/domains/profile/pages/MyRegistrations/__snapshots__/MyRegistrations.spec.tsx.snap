// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render empty state 1`] = `
<DocumentFragment>
  <section
    class="bg-theme-neutral-200"
  >
    <div
      class="px-10 py-14 bg-theme-background"
    >
      <div
        class="flex justify-between"
      >
        <div
          class="flex flex-col"
        >
          <span
            class="font-semibold text-3xl"
          >
            My Registrations
          </span>
          <span
            class="text-theme-neutral-600"
          >
            You can register a Delagate, Business and Bridgechain.
          </span>
        </div>
        <button
          class="sc-AxjAm ddQVWI self-end"
          color="primary"
          type="button"
        >
          Register
        </button>
      </div>
    </div>
    <div
      class="flex flex-col justify-center items-center mt-4 px-10 py-20 bg-theme-background"
      data-testid="my-registrations__empty-state"
    >
      <svg>
        register.svg
      </svg>
      <span
        class="mt-7 text-sm text-theme-neutral-600"
      >
        Register Business, Bridgechain and Delegate in the most convenient way.
      </span>
    </div>
  </section>
</DocumentFragment>
`;
